docs/add-runtime.md

```diff
@@ -0,0 +1,250 @@
+```
+Dockerfile - Dockerfile that explains how the container will be built.
+README.md - A readme file explaining the runtime and any special notes for the runtime. A good example of this is the PHP 8.0 runtime.
+```
+
+### 2.2 Differences between compiled and interpreted runtimes
+Runtimes within OpenRuntimes are created differently depending on whether they are compiled or interpreted. This is due to the fundamental differences between the two ways of running the code.
+
+Interpreted languages have both a `build.sh` file and a `start.sh` file.
+The `build.sh` file for an interpreted runtime is normally used for installing any dependencies for both the server itself and the user's code and then to copy it to the `/usr/code` folder which is then packaged and can be used later for running the server.
+The build script is always executed during the build stage of a function deployment.
+
+The `start.sh` file for an interpreted runtime should extract the `/tmp/code.tar.gz` file that contains both the user's code and the dependencies. This tarball was created by OpenRuntimes from the `/usr/code` folder and should install the dependencies that were pre-installed by the build stage and move them into the relevant locations for that runtime. It will then run the server ready for execution.
+
+---
+Compiled Languages only have a `build.sh` file.
+The `build.sh` script for a compiled runtime is used to move the user's source code and rename it into source files for the runtime (The `INTERNAL_RUNTIME_ENTRYPOINT` environment variable can help with this) it will also build the code and move it into the `/usr/code` folder. Compiled runtime executables **must** be called `runtime` for the ubuntu or alpine images to detect and run them.
+
+#### Note:
+`/tmp/code.tar.gz` is always created from the `/usr/code` folder as an output of the build stage. If you need any files for either compiled or interpreted runtimes you should place them there and extract them from the `/tmp/code.tar.gz` during the `start.sh` script to get the files you need.
+
+### 2.3 Writing the runtime
+Internally the runtime can be anything you like as long as it follows the standards set by the other runtimes.
+
+The best way to go about writing a runtime is like so:
+Initialize a web server that runs on port 3000 and binds to the 0.0.0.0 IP and on each `POST` request do the following:
+1. Check that the `x-internal-challenge` header matches the `INTERNAL_RUNTIME_KEY` environment variable. If not return an error with a `401` status code and an `unauthorized` error message.
+2. Decode the executor's JSON POST request. This normally looks like so:
+```json
+{
+ "env": {
+    "hello":"world!"
+ },
+ "payload":"An Example Payload",
+ "timeout": 10
+}
+```
+
+`timeout` is also an optional parameter to deal with, if you can handle it please do. Otherwise, it doesn't matter since the connection will simply be dropped by the executor.
```

Comment:
```suggestion
`timeout` is also an optional parameter to deal with, if you can handle it, please do. Otherwise, it doesn't matter since the connection will simply be dropped by the executor.
```